pipeline/scoring.py

```diff
@@ -36,12 +48,69 @@ def get_process_name(self):
 
 class Scorer:
     def __init__(self, **kwargs):
-        self.pars = ParsScorer(**kwargs)
+        self.config = Config.get()
+
+        self.pars = ParsScorer( **(self.config.value('scoring')) )
+        self.pars.augment( kwargs )
 
         # this is useful for tests, where we can know if
         # the object did any work or just loaded from DB or datastore
         self.has_recalculated = False
 
+    def score_rbbot( self, ds, deepmodel, algo=None, prov=None, session=None ):
+        if prov is None:
+            raise ValueError( "provenance can't be None" )
+        if algo is None:
+            algo = self.pars.algorithm
+
+        SCLogger.debug( "score_rbbot starting, loading cutouts data" )
+
+        sources = ds.get_sources( session=session )
+        cutouts = ds.get_cutouts( session=session )
+        cutouts.load_all_co_data( sources=sources )
+
+        # Construct the numpy array
+        # Current RBbot models assume 41×41 cutouts
+        if ( cutouts.co_dict[f'source_index_{ds.measurements[0].index_in_sources}']['sub_data'].shape
+             != (41,41) ):
+            raise ValueError( f"RBbot currently requires cutouts to be 41×41" )
+        data = np.empty( ( len(ds.measurements), 3, 41, 41 ) )
+        for i, meas in enumerate( ds.measurements ):
+            cdex = f'source_index_{meas.index_in_sources}'
+            data[ i, 0, :, : ] = cutouts.co_dict[cdex]['new_data']
+            data[ i, 1, :, : ] = cutouts.co_dict[cdex]['ref_data']
+            data[ i, 2, :, : ] = cutouts.co_dict[cdex]['sub_data']
+
+        # TODO : zero out masked pixels?
```

Comment:
TODO for this PR or future?

I'm not sure how much impact this would have on the scores if done vs not.

Reply:
I think I need to do this.  Checking with Nabeel (who build the actual RB model we're using).